noodles-bam/examples/bam_add_tag_to_header.rs

```diff
@@ -0,0 +1,73 @@
+//! Add a PG tag to the SAM header of a BAM file.
+//!
+//! This is a demonstration of how to construct a full PG entry.
+//!
+//! Verify the output by piping to `samtools view --with-header`.
+
+use std::{env, io};
+
+use noodles_bam as bam;
+use noodles_sam::header::record::value::{
+    map::{Program, Tag},
+    Map,
+};
+use noodles_sam::Header;
+
+fn add_pg(mut header: Header) -> Header {
+    const NAME: &str = "bam_add_tag_to_header";
+    const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+    // standard PG tags PN, VN, PP, CL
+    let pn = match Tag::try_from([b'P', b'N']) {
+        Ok(Tag::Other(tag)) => tag,
+        _ => unreachable!(),
+    };
+    let vn = match Tag::try_from([b'V', b'N']) {
+        Ok(Tag::Other(tag)) => tag,
+        _ => unreachable!(),
+    };
+    let pp = match Tag::try_from([b'P', b'P']) {
+        Ok(Tag::Other(tag)) => tag,
+        _ => unreachable!(),
+    };
+    let cl = match Tag::try_from([b'C', b'L']) {
+        Ok(Tag::Other(tag)) => tag,
+        _ => unreachable!(),
+    };
+    // the command-line to insert into the CL tag
+    let cmd_str = env::args().collect::<Vec<_>>().join(" ");
+
+    let program = Map::<Program>::builder().insert(pn, NAME);
+
+    let program = if let Some(last_pg) = header.programs().keys().last() {
+        program.insert(pp, last_pg.clone())
+    } else {
+        program
+    };
```

Comment:
Header records are not guaranteed to be ordered by recency. This would be incorrect for, e.g.,

```
@PG	ID:pg1	PP:pg0
@PG	ID:pg0
```